x-pack/plugins/logstash/public/services/license/logstash_license_service.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { MarkdownSimple } from '../../../../../../src/plugins/kibana_react/public';
+import { PLUGIN } from '../../../common/constants';
+
+export class LogstashLicenseService {
+  constructor(license, navigateToApp, toasts) {
+    this.license = license;
+    this.navigateToApp = navigateToApp;
+    this.toasts = toasts;
+  }
+
+  get enableLinks() {
+    return this.calculated.enableLinks;
+  }
+
+  get isAvailable() {
+    return this.calculated.isAvailable;
+  }
+
+  get isReadOnly() {
+    return this.calculated.isReadOnly;
+  }
+
+  get message() {
+    return this.calculated.message;
+  }
+
+  get isSecurityEnabled() {
+    return this.license.getFeature(`security`).isEnabled;
+  }
+
+  notifyAndRedirect() {
+    this.toasts.addDanger({
+      title: (
+        <MarkdownSimple>{this.license.getFeature(`features.${PLUGIN.ID}.message`)}</MarkdownSimple>
+      ),
+    });
+    this.navigateToApp('kibana', '#/management');
+  }
```

Comment:
In the legacy management app, "disabled" sections would still be mounted which made this necessary. With the new management app, a disabled section won't be mounted in the first place. This means there is never an opportunity for a toast message to be shown to the user that explains why they can't access this page.

I think this is ok since you could only trigger this in legacy if you went to the URL directly (there's no link in the UI). @ycombinator are you comfortable with me removing this toast, redirect code, and the message strings below?

Reply:
Yeah, I think its okay. Just one question —

> since you could only trigger this in legacy if you went to the URL directly (there's no link in the UI).
I remember specifically wanting to handle this scenario, in case a user had the URL bookmarked or something. What would be the behavior in this case in NP?

Reply:
The behavior right now is the user gets redirected to the root of the management app (similar to before). Just no toast is displayed.

Reply:
I see. And I imagine this behavior under NP (redirect w/o toast) is consistent across all management sections using NP? If so, I'm good with making this the behavior of the Logstash management section as well.